qcodes/utils/validators.py

```diff
@@ -127,13 +127,16 @@ class Bool(Validator):
 
     def __init__(self, allow_none=False):
         self._valid_values = [True, False]
+
+        self.allow_none = allow_none
```

Comment:
Would it be easier to add this to the parent class `Validator`? Then in each sub-implementation as the first step run:
`super().validate(self, value, context)`

Reply:
Maybe in this case it wouldn't be a good idea, because it basically adds this check to `validate`: `if value is None and self.allow_none: return True`

So you're basically saying, if this is the case, then it should be validated so ignore all other code. So you would still need something like:
```
return_val = super().validate(value, context)
if return_val is True:
    return True
```
so you would'nt really save lines, and it would be less clear.

Reply:
But you wouldn't have to do a check because validate raises an error if it finds a value that isn't correct.
I've had a look at the file as a whole, and to implement something like this properly you'd want to rebuild everything almost from scratch, which isn't worth it at this point.